### ISSUE TITLE
wrong sl2 offsets used

### ISSUE THREAD

**Issue by risty** [kmpm/node-sl2format#1]
File header lenght = 8 bytes(from 0 to 7). Not 10.

First byte of first frame offset = 8.
Offsets inside frame:
FrameOffset(in whole file) = 0,
LastPrimaryChannelFrameOffset = 4, //int, 4 bytes
LastDownScanChannelFrameOffset = 12, //int, 4 bytes
ThisBlockSize = 28,
PreviousBlockSize = 30,ChannelType = 32,
PacketSize = 34,
FrameIndex = 36,
UpperLimit = 40,
LowerLimit = 44,
Frequency = 50,
Depth = 64,
KeelDepth = unknown
SpeedGps = 100,
Temperature = 104,
IntLongitude = 108,
IntLatitude = 112,
WaterSpeed = 116,
CourseOverGround = 120,
Altitude = 124,
Heading = 128,
Flags = 132,
TimeOffset = 140,
SoundedData = 144

**Comment by kmpm** [comment]
Where did you get this information? It's not that I doubt you, just curious.

I'm going to test this and see what happens.

**Comment by risty** [comment]
thread with 500 pages at russian fishing forum. do you need a link? )))
i'm implementing small api for sl2/sl3 at c# now and decide to send you debug data.
update about first half of frame:
FrameOffset = 0,//int, 4 bytes
		LastPrimaryChannelFrameOffset = 4, //int, 4 bytes
		LastSecondaryChannelFrameOffset = 8, //int, 4 bytes
		LastDownScanChannelFrameOffset = 12, //int, 4 bytes
		LastSidescanLeftChannelFrameOffset = 16, //int, 4 bytes
		LastSidescanRightChannelFrameOffset = 20, //int, 4 bytes
		LastSidescanCompositeChannelFrameOffset = 24, //int, 4 bytes
		ThisBlockSize = 28, //Bytes to next frame Start
		PreviousBlockSize = 30, //Bytes to previous frame Start 

TimeOffset seems like milliseconds from file creation time.

**Comment by kmpm** [comment]
You don't have documentation about the header as well? (fingers crossed)

**Comment by risty** [comment]
fileversion = 0//short, 2 bytes  1/SLG, 2/SL2, 3/SL3
HardwareVersion? =2 //short, 2 bytes. 0/Button? 1/Touch?
BlockSize = 4 //short, 2 bytes. 1970/Downscan #b207, 3200/Sidescan #800c
bytes  6 and 7 always zero at all logs i had seen.

**Comment by kmpm** [comment]
I'm skeptical about hardwareversion. 
I'v got 0 from a HDS 7 and 1 from a Elite 4 CHIRP.

**Comment by risty** [comment]
agree with you. all i had

**Comment by kmpm** [comment]
What do you think about the frequency field, Single byte or short (UInt16)?

**Comment by risty** [comment]
single byte,
```
public enum Frequency : byte
	{
		Frequency_200KHz = 0,
		Frequency_50KHz = 1,
		Frequency_83KHz = 2,
		Frequency_455KHz = 3,
		Frequency_800KHz = 4,
		Frequency_38KHz = 5,
		Frequency_28KHz = 6,
		Frequency_130KHz210KHz = 7,
		Frequency_90KHz150KHz = 8,
		Frequency_40KHz60KHz = 9,
		Frequency_25KHz45KHz = 10,
	}
```

**Comment by risty** [comment]
https://github.com/risty/SonarLogApi
full sl2 and partially sl3 offsets here - https://github.com/risty/SonarLogApi/blob/master/SonarLogAPI/Lowrance/Frame.cs

**Comment by Chris78** [comment]
Hi all,
I wrote a short Ruby script that converts sl2 files into textfiles (csv for example). The offsets can be adopted easily for testing. 
I successfully retrieved GPS coordinates, depths, temperature etc. from my sl2 files, recorded by a Lowrance Hook 4. I also included some conversion into common decimal degree coordinates.
https://github.com/Chris78/sl2decode
Maybe it's of use for somebody.

Chris
